test/runnable/test18772.d

```diff
@@ -8,5 +8,8 @@ void main()
     cfloat[1] A;
     float[1] B;
     int i = 0;
+    A[0] = 2.0f + 4i;
+    B[0] = 3.0f;
+    assert(((2.0f + 4i) * 3.0f).re == fun(A[i] * B[i]));
```

Comment:
I would add another function that returns the imaginary part, I suspect that will be broken.
Also, passing the cfloat the second or higher parameter will need to place it in other registers, I guess this will also be a problem.
What about cdouble/creal?

Reply:
Adding the second function for the imaginary part reveals that the imaginary part is not getting copied.